### ISSUE TITLE
Bug: RapiDoc and Stoplight Elements fail when used as root schema

### ISSUE THREAD

**Issue by tcbegley** [litestar-org/litestar#3047]
### Description

Setting `root_schema_site="rapidoc"` or `root_schema_site="elements"` in the `OpenAPIConfig` results in an error when visiting `/schema`. Both `/schema/rapidoc` and `/schema/elements` work as expected.

The issue appears to be related to the fact that both RapiDoc and StopLight elements [are configured with `spec-url`](https://github.com/litestar-org/litestar/blob/747fb90772721e7867f2e1ff2da6636f8442a6fb/litestar/openapi/controller.py#L494) whereas Swagger UI and Redoc HTML has the OpenAPI JSON inlined via a call to `OpenAPIController._get_schema_as_json`.

### MCVE

```python
from litestar import Litestar, get
from litestar.openapi import OpenAPIConfig


@get("/")
async def hello_world() -> str:
    return "Hello world!"


app = Litestar(
    route_handlers=[hello_world],
    openapi_config=OpenAPIConfig(
        title="Test app", version="0.0.1", root_schema_site="rapidoc"
    ),
)
```

### Steps to reproduce

```bash
1. Run app with `litestar run`
2. Visit `localhost:8000/schema` in your browser.
```

### Litestar Version

2.5.0

### Platform

- [X] Linux
- [X] Mac
- [ ] Windows
- [ ] Other (Please specify in the description above)

**Comment by provinzkraut** [comment]
Can you update to the latest version (`pip install -U litestar==2.5.3`) and confirm this issue persists?

**Comment by tcbegley** [comment]
Upgraded, and unfortunately the issue persists with `2.5.3` installed.

**Comment by JacobCoffee** [comment]
Using the provided MCVE I don't seem to be able to reproduce :\
```
~ via 🐋 colima via  pyenv
➜ ntp -v 3047
Directory /tmp/testing/3047 created and switched to.
Virtual environment created and activated.

/tmp/testing/3047 via  pyenv (.venv)
➜ vim mcve.py

/tmp/testing/3047 via  pyenv (.venv) took 3s
➜ pip install litestar[standard]

/tmp/testing/3047 via  pyenv (.venv) took 6s
➜ litestar run --reload --reload-dir . --debug


/tmp/testing/3047 via  pyenv (.venv) took 6s
➜ browse 127.0.0.1/schema
```
result:
<img width="1201" alt="image" src="https://github.com/litestar-org/litestar/assets/45884264/a39060ad-0e03-4d95-8172-791564cc71bd">
and same for Stoplight.

tested on release in 2.5 - 2.6.

Am i doing something dumb here?

**Comment by tcbegley** [comment]
Wow, that's really strange... I just tested again at my end with a clean virtual environment and the above example and saw the same issue.

```
pip install -U "litestar[standard]"
litestar run --reload --reload-dir . --debug
open http://127.0.0.1:8000/schema
```

The request is being received by the backend as expected

```
INFO:     127.0.0.1:55586 - "GET /schema HTTP/1.1" 200 OK
INFO:     127.0.0.1:55586 - "GET /openapi.json HTTP/1.1" 404 Not Found
ERROR - 2024-02-11 10:46:25,461 - litestar - config - exception raised on http connection to route /openapi.json
```

Issue does appear to be `/openapi.json` route as if I visit `/schema/rapidoc` instead I get

```
INFO:     127.0.0.1:55628 - "GET /schema/rapidoc HTTP/1.1" 200 OK
INFO:     127.0.0.1:55628 - "GET /schema/openapi.json HTTP/1.1" 200 OK
```

I tried private windows and force refreshing the page but always the same result

<img width="864" alt="image" src="https://github.com/litestar-org/litestar/assets/15220906/6aae1341-07b9-453c-9883-46ed004bc1be">

In case it helps I tested just now with Python 3.11.4 and this is the result of `pip freeze`.
[pip-freeze.txt](https://github.com/litestar-org/litestar/files/14232018/pip-freeze.txt)

No idea what's going on but I'm also very open to the possibility that I'm doing something dumb on this end 😅

**Comment by provinzkraut** [comment]
@tcbegley I also can't reproduce this. Are you sure that you're running the correct application file?

**Comment by tcbegley** [comment]
Hey @provinzkraut, thanks for also looking into this.

I noticed from @JacobCoffee's screenshot that he has `/schema/` in his address bar whereas I have `/schema`. Sure enough if I visit `/schema/` then I don't get the error. By contrast `/schema/rapidoc` works but `/schema/rapidoc/` does not. I guess an issue with relative paths resolving to the wrong place? Ditto for StopLight Elements. Are you able to reproduce by visiting `/schema`?

To answer your original question, definitely running the right application file. To double check I added

```python
if __name__ == "__main__":
    import uvicorn
    uvicorn.run(app)
```

to the bottom of the above example and ran `python app.py` and observed the same result.

**Comment by provinzkraut** [comment]
Yes, I was able to reproduce this with the help of @Alc-Alc who came to the same conclusion as you.

I'm not yet sure if this is an upstream issue or a bug on our side, but he provided some additional insight:

```
INFO:     127.0.0.1:49743 - "GET /schema HTTP/1.1" 200 OK
INFO:     127.0.0.1:49743 - "GET /openapi.json HTTP/1.1" 404 Not Found
INFO:     127.0.0.1:49743 - "GET /schema HTTP/1.1" 200 OK
INFO:     127.0.0.1:49743 - "GET /schema/openapi.json HTTP/1.1" 200 OK
```

these are the paths requested with/without the trailing slash, so something in there is not building them correctly.

**Comment by guacs** [comment]
@tcbegley I think #3196 should fix this. Could you confirm this?

**Comment by tcbegley** [comment]
Hey @guacs, I installed from #3196 and can confirm it resolved the issue for me. Thanks!
